h2o-automl/src/main/java/ai/h2o/automl/Models.java

```diff
@@ -0,0 +1,64 @@
+package ai.h2o.automl;
+
+import hex.Model;
+import hex.ModelContainer;
+import water.*;
+import water.util.ArrayUtils;
+
+import java.lang.reflect.Array;
+import java.util.Arrays;
+
+public class Models<M extends Model> extends Lockable<Models<M>> implements ModelContainer<M> {
+
+    private final Class<M> _clz;
+    private final Job _job;
+    private Key<M>[] _modelKeys = new Key[0];
+
+    public Models(Key<Models<M>> key, Class<M> clz) {
+        this(key, clz, null);
+    }
+
+    public Models(Key<Models<M>> key, Class<M> clz, Job job) {
+        super(key);
+        _clz = clz;
+        _job = job;
+    }
+
+    @Override
+    public Key<M>[] getModelKeys() {
+        return _modelKeys.clone();
+    }
+
+    @Override
+    public M[] getModels() {
+        return Arrays.stream(_modelKeys)
```

Comment:
if you have many objects, it is better to call first `DKV.prefetch(key)` in a first loop - this will start transfer of all the objects in parallel

the second loop (DKV.get) then just waits for the transfer to finish

Reply:
nice tip, will change my implementation, thx